server/api.js

```diff
@@ -1,78 +1,207 @@
- router.get('/callback', async (req, res) => {
-    const { code } = req.query;
-    const options = {
-      code,
-    };
-
-    try {
-	  const accessToken = await client.getToken(options);
-		console.log('acces token', accessToken)
-	  //accessing the token number from the above
-	  const token = accessToken.token.access_token; 
-
-      //authenticates the access_token sent by github during the Oauth2 flow
-	  const octokit = new Octokit({
-		auth:token,
-	  });
-
-	  //this returns the authenticated user's username/login
-	  const { data } = await octokit.request("/user");
-	  return res.status(200).json(data.login);
-
-	
-    } catch (error) {
-      console.error('Access Token Error', error.message);
-      return res.status(500).json('Authentication failed');
+//const githubName = "Buchrateka1984";
+
+router.get("/", (_, res, next) => {
+  Connection.connect((err) => {
+    if (err) {
+      return next(err);
     }
+    res.json({ message: "Hello, world!" });
   });
+});
+router.get("/graduates", (_, res, next) => {
+  Connection.connect((err) => {
+    if (err) {
+      return next(err);
+    }
+    Connection.query("SELECT * FROM graduates", (error, result) => {
+      res.json(result.rows);
+    });
+  });
+});
+// create new profile
+router.post("/graduates", function (req, res) {
+  const newFirstName = req.body.first_name;
+  const newSurname = req.body.surname;
+  const github_name = req.body.githubName;
+  console.log(newFirstName, newSurname, github_name);
+  Connection.query(
+    `SELECT * FROM github_accounts WHERE account_name = $1`,
+    [github_name],
+    (err, result) => {
+      console.log(result.rowCount);
+      if (result.rowCount > 0) {
+        const github_id = result.rows[0].id;
+        console.log(github_id);
+        Connection.query(
+          `insert into graduates (first_name, surname, github_id) values` +
+            `($1,$2,$3)`,
+          [newFirstName, newSurname, github_id],
+          (error, result) => {
+            res.json(result.rows);
+          }
+        );
+      } else
+        res.send({
+          message: "this is  github account does not belong to a CYF graduates",
+        });
+    }
+  );
+});
+//checking the github username exist in our database
+router.get("/accounts/:name", (req, res, next) => {
+  const githubName = req.params.name;
+  Connection.connect((err) => {
+    if (err) {
+      return next(err);
+    }
+    Connection.query(
+      "SELECT * FROM github_accounts where account_name=$1 ",
+      [githubName],
+      (error, result) => {
+        if (result.rowCount > 0) {
+          Connection.query(
+            "SELECT * FROM github_accounts GA join graduates G on(GA.id=G.github_id) where GA.account_name=$1 ",
+            [githubName],
+            (error, result) => {
+              if (result.rowCount > 0) res.json(result.rows);
+              else
+                res.send({ account_name: githubName, profile_status: false });
```

Comment:
you don't need this one if you use LEFT JOIN, if there's no graduate in this case github_id will be null